lighthouse-core/report/v2/renderer/details-renderer.js

```diff
@@ -134,3 +197,21 @@ DetailsRenderer.ListDetailsJSON; // eslint-disable-line no-unused-expressions
  * }}
  */
 DetailsRenderer.CardsDetailsJSON; // eslint-disable-line no-unused-expressions
+
+/** @typedef {{
+ *     type: string,
+ *     header: ({text: string}|undefined),
+ *     items: !Array<!Array<!DetailsRenderer.DetailsJSON>>,
+ *     itemHeaders: !Array<!DetailsRenderer.DetailsJSON>
```

Comment:
by letting these be `DetailsJSON` we're basically descending into recursive madness again since in theory when you call `this.render` on them they could embed a whole other table (but of course, our css would quickly fall over itself if that sort of thing occurred). This leaves these basically untyped (except our assurance to the compiler that we're correct when we cast in `this.render`)

I'm not sure of the right solution...maybe we could separate top level rendering (anything currently in `this.render`) from more primitive-level rendering (just text, url, and thumbnail as of this PR) in a future PR or something, but we would need to make whatever we did non-annoying.